### ISSUE TITLE
Is the learning rates bound parameter actually used in the code?

### ISSUE THREAD

**Issue by vvanirudh** [NVlabs/diff-dope#10]
I noticed that the config parameter is read [here](https://github.com/NVlabs/diff-dope/blob/main/diffdope/diffdope.py#L1374-L1381) in the code, but I do not see `self.learning_parameters` being used anywhere else in the code (especially in `run_optimization()` where we simply [use](https://github.com/NVlabs/diff-dope/blob/main/diffdope/diffdope.py#L1664-L1670) `base_lr` with decay). However, the algorithm in the paper clearly states that the randomly sampled learning rates are what is being used for optimization. Any reason why there is a discrepancy?

**Comment by TontonTremblay** [comment]
Yeah it is being used correctly, https://github.com/NVlabs/diff-dope/blob/main/diffdope/diffdope.py#L1374

**Comment by vvanirudh** [comment]
But where is `self.learning_rates` being used in `run_optimization` (or anywhere else). The line you linked initializes the sampled learning rates, but I don't see them being used anywhere

**Comment by vvanirudh** [comment]
Ah I figured it out. It is directly applied on the loss
